common/LCSTextDifferencer.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.common;
+
+import com.google.startupos.common.TextChange.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.swing.text.Segment;
+
+/** An implementation of text difference based on the Longest Common Subsequence (LCS) problem. */
+public class LCSTextDifferencer implements TextDifferencer {
```

Comment:
I see you kept the `TextDifferencer` interface.
If we only have one implementation, we don't need an interface. Not having an interface also saves us from needing a provider, to connect the interface to the class.